src/helpers.js

```diff
@@ -12,6 +12,44 @@ import { processJSONLogicNode } from './jsonLogic';
 import { hasProperty } from './utils';
 import { buildCompleteYupSchema, buildYupSchema } from './yupSchema';
 
+/**
+ * List of attributes for each JSF' field
+ * that do not exist in JSON Schema directly
+ * */
+const fieldAttrsFromJsf = [
+  'name', // driven from json schema properties' name
+  'label', // driven from schema field title
+  'required', // driven from schema conditionals
+  'type', // Type is @deprecated, but keep it until fully removed. Otherwise E2E tests (private) fail.
+  'inputType', // driven from json type + presentation
+  'jsonType', // driven from json type
+  'getComputedAttributes', // for json-logic
+  'computedAttributes', // From json-logic
+  'calculateConditionalProperties', // driven from conditionals
+  'calculateCustomValidationProperties', // To be deprecated in favor of json-logic
+  'schema', // Yup schema validator based on json validations
+  'scopedJsonSchema', // The respective JSON Schema,
+  'isVisible', // Driven from conditionals state
+];
+const fieldJsfAttrsObj = Object.fromEntries(fieldAttrsFromJsf.map((k) => [k, true]));
+
+/**
+ *
+ * @param {*} finalField
+ * @param {*} branchAttrs
+ */
+function removeStaleAttrsFromField(field, latestAttrs) {
+  Object.keys(field).forEach((key) => {
+    if (
+      latestAttrs[key] === undefined &&
+      fieldJsfAttrsObj[key] === undefined // ignore these because they are internal
```

Comment:
note: In the original PR (https://github.com/remoteoss/json-schema-form/pull/52) it was created a list of attrs that can be removed. In this PR, I think it's safer to do it the other way around: create a list of fields that cannot be removed, which is basically every internal computed JSF attr (`fieldJsfAttrsObj`).

I still don't think this is the ideal solution and the real problem is somewhere earlier in the data flow.

But based on the urgency to unblock other features/consumers and the strong unit-tests, I'm comfortable with this solution.